materials/08_statistical_analysis_all_stages.Rmd

```diff
@@ -0,0 +1,417 @@
+:::
+
+```{r, eval=TRUE, include=FALSE}
+library("QFeatures")
+library("NormalyzerDE")
+library("limma")
+library("factoextra")
+library("org.Hs.eg.db")
+library("clusterProfiler")
+library("enrichplot")
+library("patchwork")
+library("tidyverse")
+library("pheatmap")
+library("here")
+
+load(here("materials/output/lesson04.rda"), verbose = TRUE)
+```
+
+## Differential expression analysis
+
+Having cleaned our data, aggregated from PSM to protein level, completed
+a log2 transformation and normalised the data, we are now ready to carry
+out statistical analysis. 
+
+The aim of this section is to answer the
+question: *"Which proteins show a significant change in abundance
+between the multiple cell cycle stages?"*.
+
+Our null hypothesis is: **(H0)** The change in abundance for a protein
+between cell cycle stages is zero.
+
+Our alternative hypothesis is: **(H1)** The change in abundance for a
+protein between cell cycle stages is not zero.
+
+
+We will use `limma` to perform the statistical tests using an **empirical Bayes-moderated linear model**. Please see the [Statistical analysis](./06_statistical_analysis.html) section for more details about `limma` and its application in quantitative proteomics. 
+
+Depending on whether the linear model is used to perform single comparisons or multifactorial comparisons, the test statistic will either be a
+t-value or an F-value, respectively. Here, we are performing a multifactorial comparison, so we will obtain an F-value. 
+
+### What is an F-value?
+
+An F-value is a parametric statistical value used to compare the mean
+values of **three or more** groups. The F-value is the ratio of the between-group
+variation (explained variance) and the within-group variance (unexplained variance).
+The higher the F-value, the more significant the difference between the groups is.
+
+### How is the p-value obtained?
+
+A p-value may be obtained from an F-value by comparing the value against an F-distribution with the appropriate degrees of freedom.
+    
+-   **degrees of freedom** = the number of observations minus the number
+    of independent variables in the model
+-   **p-value** = the probability of achieving the F-value under the
+    null hypothesis i.e., by chance
+    
+Here, we will perform a comparison between three groups (M phase, G1 phase and
+desynchronised) for each protein and obtain an F-value and p-value for each protein.
+
+First, we extract the protein assay data. We will remove the pre-treatment sample,
+since this condition is not replicated and so not amenable to statistical testing.
+```{r}
+
+## Identify the columns for the Pre-treatment sample in the protein assay
+pre_treatment_col_ix <- colData(cc_qf[["log_norm_proteins"]])$condition == 'Pre-treatment'
+
+# Extract the protein assay from our QFeatures object, excluding the pre-treatment sample
+all_proteins <- cc_qf[["log_norm_proteins"]][, !pre_treatment_col_ix]
```

Comment:
Same comment as in the other notebook about possibly simplifying the code here so it doesn't have as many chained `[[`, `[`, `$`

Reply:
Agreed. Be thankful I didn't do it in one line 😆